### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by stemy2** [Tobi823/ffupdater#38]
Hello,

The app crash when trying to find an update to Iceraven. Here is the log:

```
de.marmaro.krt.ffupdater.MainActivity$UpdateCheckException at MainActivity.kt:199
de.marmaro.krt.ffupdater.MainActivity$UpdateCheckException

de.marmaro.krt.ffupdater.MainActivity$UpdateCheckException: fail to check ICERAVEN for updates
	at de.marmaro.krt.ffupdater.MainActivity$checkForAppUpdate$1.invokeSuspend(MainActivity.kt:199)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: de.marmaro.krt.ffupdater.app.impl.exceptions.ApiNetworkException: Fail to consume API. Previous exceptions: [https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest; https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest; https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest; https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest; https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest]. Last exception:
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.readNetworkResourceWithRetries(ApiConsumer.kt:62)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer$readNetworkResourceWithRetries$1.invokeSuspend(Unknown Source:12)
	... 6 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:251)
	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getInputStream(DelegatingHttpsURLConnection.java:210)
	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getInputStream(Unknown Source:0)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.readNetworkResource(ApiConsumer.kt:73)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.readNetworkResourceWithRetries(ApiConsumer.kt:55)
	... 7 more


Android Version: 27
Device Manufacturer: samsung
Device Model: SM-G930F
```

**Comment by Tobi823** [comment]
I guess it's a temporary network issue.

Just to be sure - can you open https://api.github.com/repos/fork-maintainers/iceraven-browser/releases/latest ?

**Comment by Tobi823** [comment]
I think this is the best solution: If it is likely that it is just a network error, then display "No network connection" instead of crashing

**Comment by Tobi823** [comment]
Should be fixed in version 73.1.2
